code/datums/circuits.dm

```diff
@@ -0,0 +1,122 @@
+	var/global/list/assigned_boards = list()
+	//Each bitflag points to a board!
+	var/localbit = 0 //What are WE programmed to? Always start as 0
+
+	var/table_options = " align='center'"
+	var/row_options1 = " width='80px'"
+	var/row_options2 = " width='80px'"
+	var/window_x = 240
+	var/window_y = 300
+
+/datum/circuits/New(var/atom/homeboard)
+	..()
+	holder = homeboard
+	if(!(src.assigned_boards.len))
+		generate_schema()
+
+/datum/circuits/proc/generate_schema()
+	for(var/C in possible_boards)
+		var/newbit = newhash(CHOOSE_FUSES)
+		while(!check_config(newbit))
+			newbit = newhash(CHOOSE_FUSES)
+		assigned_boards["[newbit]"] = C
+	return
+
+/datum/circuits/proc/check_config(var/proposed)
+	for(var/bitflag in assigned_boards)
+		if(text2num(proposed) == text2num(bitflag))
+			return 0
+	return 1
+
+/datum/circuits/proc/newhash(var/choose) //Returns a bitflag
+	var/list/fuse_point_list = list(ALPHA,BETA,GAMMA,DELTA,ETA,THETA,IOTA)
+	var/build = 0
+	var/choice = null
+	while(choose>0)
+		choice = pick_n_take(fuse_point_list)
+		build |= choice
+		choose--
+	return build
+
+//The greek being passed here is one of those previously mentioned defined variables. Each one is its own binary (e.g.: BETA = 2, GAMMA = 4)
+
+/datum/circuits/proc/checkfuse(var/greek)
+	return localbit & greek  //true if any bits in `localbit` and `greek` overlap, ie any are set in both
+                             //this is because it returns the "intersection" (of the 2 sets), ie 1101 & 1010 returns 1000
+
+/datum/circuits/proc/togglefuse(var/greek)
+	localbit ^= greek //The ^= uses XOR and it basically toggles the given bits
+                  //eg 1111 ^ 1000 = 0111
+                  //eg 0000 ^ 1000 = 1000
+	return
+
+/datum/circuits/proc/Interact(var/mob/living/user)
+	if(!istype(user))
+		return 0
+	var/html = null
+	if(holder)
+		html = GetInteractWindow()
+	if(html)
+		user.set_machine(holder)
+	var/datum/browser/popup = new(user, "circuits", holder.name, window_x, window_y)
+	popup.set_content(html)
+	popup.set_title_image(user.browse_rsc_icon(holder.icon, holder.icon_state))
+	popup.open()
+
+/datum/circuits/proc/GetInteractWindow()
+	var/html = "<div class='block'>"
+	html += "<h3>Protoboard</h3>"
+	html += "<table[table_options]>"
+
+	for(var/fusepoint in fuse_point_names)
+		html += {"<tr>
+		<td[row_options1]><font color='blue'>[fusepoint]</font>
+		</td>
+		<td[row_options2]>
+		<A href='?src=\ref[src];action=1;fuse=[fuse_point_names[fusepoint]]'>[checkfuse(text2num(fuse_point_names[fusepoint])) ? "Melt" :  "Fuse"]</A></td></tr>"}
+	html += "</table>"
+	html += "</div>"
+
+	return html
+
+/datum/circuits/Topic(href, href_list)
+	..()

```

Comment:
this wont do nuffin because /datum doesn't have a base topic like /obj/machinery does